### ISSUE TITLE
Fenced codeblock not recognizes in wysiwyg editor when indented

### ISSUE THREAD

**Issue by simoneggler** [jonathanyeung/mark-sharp#21]
### Describe the bug.

codeblocks that are indented (e.g. in a unnumbered list) are not recognized as such in the wysiwyg mode

### Repro Steps

1. create a codeblock with indents
e.g. 
<img width="82" alt="image" src="https://github.com/user-attachments/assets/8a612366-f851-41f6-b9a2-0ce7153a9560">

3. change to wysiwyg mode

**Comment by jonathanyeung** [comment]
Hi @simoneggler - can you paste the full markdown definition?  Are you using tabs or spaces for indentation? If spaces, how many spaces are your indentations?

Per the GFM specification (https://github.github.com/gfm/#list-items), the indentation needs to go to the first non-whitespace character that isn't part of the list marker.

This is what I'm seeing (raw markdown on left, mark sharp on right):

<img width="1051" alt="image" src="https://github.com/user-attachments/assets/2e1974d7-373b-4123-94ac-20cb1507c46d">

**Comment by simoneggler** [comment]
Hmm, your example renders fine in the vscode preview, but not in your wysiwig editor:

example (not escaped, it's harde to escape code fences:

1. first item
    ```
    foobar
    ```
1. second item

renders like this: 
<img width="205" alt="image" src="https://github.com/user-attachments/assets/410a89dd-67d7-419c-b0dd-68e6963a0403">

**Comment by simoneggler** [comment]
as you can see, it renders nicely in the github comment too

**Comment by jonathanyeung** [comment]
Hi @simoneggler - I think I've identified the issue. It has to do with line-ending type. If the document is set to CRLF line endings, then I can reproduce the issue.  I believe this is actually the same root cause for the other issue you filed (https://github.com/jonathanyeung/mark-sharp/issues/20).

I should have a fix for this in the next release.  As a temporary workaround, VS Code lets you toggle between CRLF/LF line endings. If you toggle to LF style endings, it should parse correctly.

<img width="559" alt="image" src="https://github.com/user-attachments/assets/782d06b3-2170-48a0-83d6-950fe0a52698">


I'll loop back on this thread once a fix has been released.  Thanks for filing this! It's a pretty important problem that you've identified.
